### DISCUSSION TITLE
An unhandled exception occurred while processing the request.

### DISCUSSION THREAD

**Discussion by csilipo** [aspnetrun/run-aspnetcore-microservices#42] [General]
Hi, 
Thanks for the Course - I just wanted to test out the Github code. I got this Exception error below:

Just one question. How do I fix this error?

thanks
carlo
ps I can goto Catalog API -> http://host.docker.internal:8000/swagger/index.html 'WORKS'
and most of the other containers seems to be working aswell.

------------------------------------------------------------------------------------------------------------------------------------

An unhandled exception occurred while processing the request.
NullReferenceException: Object reference not set to an instance of an object.
AspnetRunBasics.Pages.Shared.Pages_Shared__TopProductPartial.ExecuteAsync() in _TopProductPartial.cshtml, line 8

Stack Query Cookies Headers Routing
NullReferenceException: Object reference not set to an instance of an object.
AspnetRunBasics.Pages.Shared.Pages_Shared__TopProductPartial.ExecuteAsync() in _TopProductPartial.cshtml
Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageCoreAsync(IRazorPage page, ViewContext context)
Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageAsync(IRazorPage page, ViewContext context, bool invokeViewStarts)
Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderAsync(ViewContext context)
Microsoft.AspNetCore.Mvc.TagHelpers.PartialTagHelper.RenderPartialViewAsync(TextWriter writer, object model, IView view)
Microsoft.AspNetCore.Mvc.TagHelpers.PartialTagHelper.ProcessAsync(TagHelperContext context, TagHelperOutput output)
Microsoft.AspNetCore.Razor.Runtime.TagHelpers.TagHelperRunner.<RunAsync>g__Awaited|0_0(Task task, TagHelperExecutionContext executionContext, int i, int count)
AspnetRunBasics.Pages.Pages_Index.ExecuteAsync() in Index.cshtml

**Comment by nealu** [comment]
How did you get anything to run based on link:
 http://host.docker.internal:8000/swagger/index.html
